chromium_src/third_party/blink/common/web_preferences/web_preferences_mojom_traits.cc

```diff
@@ -25,6 +25,22 @@ bool StructTraits<blink::mojom::WebPreferencesDataView,
   out->force_cosmetic_filtering = data.force_cosmetic_filtering();
   out->hide_media_src_api = data.hide_media_src_api();
   out->should_detect_media_files = data.should_detect_media_files();
+  out->allow_to_run_script_on_main_world =
+      data.allow_to_run_script_on_main_world();
+
+  mojo::MapDataView<mojo::StringDataView, mojo::StringDataView> data_view;
+  data.GetUrlAndMediaDetectionScriptsDataView(&data_view);
+
+  for (size_t i = 0; i < data_view.size(); ++i) {
+    std::string key;
+    std::string value;
+    if (!data_view.keys().Read(i, &key) ||
+        !data_view.values().Read(i, &value)) {
+      continue;
+    }
+    out->url_and_media_detection_scripts.insert({key, value});
```

Comment:
passing the scripts like this doesn't seem like a great idea as I mentioned above, but this kind of thing should have had tests

Reply:
in fact it looks like there already is one so you can just do all of this in `OnMediaUpdatedFromContents`. There's no need for any new renderer code here

Reply:
https://github.com/brave/brave-core/pull/21332#discussion_r1458087646

Moved back to browser side.